server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java

```diff
@@ -2063,17 +2065,23 @@ private boolean isHashJoinEnabled() {
     private RandomizedRulesConfig randomizedRulesConfig() {
         var useRandomizedOptimizerRules = getTestAnnotation(UseRandomizedOptimizerRules.class);
         var enabled = false;
+        double amountOfRulesToDisable = 0.0d;
         List<Class<? extends io.crate.planner.optimizer.Rule<?>>> rulesToKeep = List.of();
         if (useRandomizedOptimizerRules != null) {
             enabled = isFeatureEnabled(useRandomizedOptimizerRules.value());
             if (enabled) {
                 rulesToKeep = Arrays.asList(useRandomizedOptimizerRules.alwaysKeep());
+                if (useRandomizedOptimizerRules.disable() == -1) {
```

Comment:
Please add an assertion that value is either -1 or [0.1, 1].

Reply:
That's handled already in `buildRandomizedRuleSessionSettings` but we can move it here.

Reply:
I think it's more clear here, since it's the "entry point" from the annotation.

Reply:
For `buildRandomizedRuleSessionSettings` the assertion is needed to work correctly.

Reply:
However, we can be generous with assertions since they are removed at runtime anyway.

Reply:
it's only tests here, not production code, and the assertion there is:
```
assert factor >= 0 && factor <= 1 : "Factor for Rule Randomization must between 0 and 1";
```, so it allows 0, but it doesn't make sense to have 0, right? (I mean from a usage percepective of the annotation, if you want to not disable anything you just set the value = 0).

Reply:
So you want two assertions?

Reply:
or just here, I think it's valuable to not allow for example -2, or 2 or 0 from the annotation, as if a dev accidentally puts something like that, it maybe confusing of how a test behaves after all.

Reply:
The point is that `buildRandomizedRuleSessionSettings` can break if we remove the assertion there. Here is there is nothing what can break.

Reply:
ok, It's fine, the `0` though imho shouldn't be allowed, as it negates the value of `value` from the annotation, but It's minor, no need to waste your time more on this, apologies.